### PR TITLE
CLIC.Subcommand.Instance: Add a procedure to load aliases from config

### PR DESCRIPTION
**Pull request by Fabien-Chouteau**

### CODE REVIEW

File: src/clic-subcommand-instance.ads [hunk 1]
```diff
@@ -42,6 +44,15 @@ package CLIC.Subcommand.Instance is
...
    --  If Alias is already set, it will be silently replaced.
 
+   procedure Load_Aliases (Conf     : CLIC.Config.Instance;
+                           Root_Key : String := "alias");
+   --  Load aliases from the given configuration.
+   --
+   --  All the config keys in the format "<Root_Key>.<sub-key>" are
+   --  processed. If the value of a key is a not null String, <sub-key>
+   --  is registered as an alias for the value (parsed into a list with
+   --  GNAT.OS_Lib.Argument_String_To_List).
```
Comment (mosteo) [inline src/clic-subcommand-instance.ads:54]: I'm having trouble visualizing this... Maybe having an example here would help?

**Review by mosteo — APPROVED** [review]

File: src/clic-subcommand-instance.ads (hunk 1 above)
Comment (Fabien-Chouteau) [reply src/clic-subcommand-instance.ads]: I tried to make it clearer.

File: src/clic-subcommand-instance.ads [hunk 2]
```diff
@@ -42,6 +44,23 @@ package CLIC.Subcommand.Instance is
...
+   --
+   --  For instance when Root_Key := "alias" (default):
+   --  alias.test1 = "cmd arg1 arg2" # Valid: test1 -> ["cmd", "arg1", "arg2"]
+   --  alias.test2 = "cmd"           # Valid: test2 -> ["cmd"]
+   --  alias.test3 = ""              # Ignored
+   --  alias.test4.test = "cmd arg1" # Ignored
+   --  alias.test5 = 20              # Ignored
+   --  alias.test6 = ["cmd", "arg1"] # Ignored
```
Comment (mosteo) [inline src/clic-subcommand-instance.ads:62]: Perfectly clear now, thanks! But the examples make me ponder: wouldn't the `test6` syntax be safer from a point of view of spaces in arguments?

Comment (Fabien-Chouteau) [reply src/clic-subcommand-instance.ads:62]: Indeed, but the current implementation for `CLIC.Config` doesn't support array as config value.

Comment (mosteo) [reply src/clic-subcommand-instance.ads:62]: Ah right.

**Pull request merged**